lib/ui/painting.dart

```diff
@@ -3856,6 +3862,9 @@ enum ClipOp {
   intersect,
 }
 
+/// A flag to enable use of the new DisplayList format for Picture objects.
+bool _useDisplayListPictures = false;
```

Comment:
I'm torn on how to do this. I'd like to have a programmatic way to flip this at runtime so I can write tests more easily that depend on one or the other. But, that would create a new public API.

I do want to hook this up to a command-line flag, though, which sort of creates a public API anyway. We already have plans to do that next.

Is bool.fromEnv() available on all platforms? Could I do an A/B on a phone with that and setting a shell env variable?

Reply:
FYI, I've made the "which way was it created?" conundrum a little more deterministic in the latest edits I'm working on by having only Recorder look at it. Canvas will then create a canvas to match the recorder rather than relying on the boolean. This prevents creating an inconsistent Recorder/Canvas pair.

Reply:
bool.fromEnvironment will read a dart define which is specified on the command line.  We don't normally treat those as stable or part of public API.

Reply:
FWIW, dart defines are how we control whether to use CanvasKit or the HTML backend on web, which this feels somewhat analogous to.

Reply:
There has been some trouble trying to use <type>.fromEnv() in flutter code. I sent a message to @jonahwilliams asking about this. The standard theory is that you can use that, but it has to be a const so it gets compiled in, but that doesn't seem to be working right now. Also, I'd like this to be decided at app run time so the const may not meet those needs. It does seem to work from the web code - but that is compiled differently than the mobile engine code.

Reply:
We cannot use fromEnvironment in engine code today, unless we build separate engines for each state. (see how we handle web debug mode today). These working in the web engine release mode is a bit of a coincidence .

Instead, you could set a define and then have the engine read the value from the snapshot, and use a native call to determine true/false.

Reply:
Sgtm

Reply:
This doesn't sound good to me. We may not be able to do this today, but what would it take to achieve this?